### DISCUSSION TITLE
[RFC] Make API docs a bit more "social"

### DISCUSSION THREAD

**Discussion by byCedric** [expo/expo#8765] [Proposals and Ideas]
Original idea by @ccheever ❤️ 

## Why

Developers usually have different use cases in what and how they want to implement certain APIs. While the API docs cover the most common ones and provide additional context, not all use cases are handled. If you look at the PHP API docs, you can see a section with "User Contributed Notes". This section contains some additional info that other users found useful when using this. If we want to make the API docs a "go-to" for all developers, this could be a good thing to implement.

## Proposal

 Currently, @ccheever and I thought of an option to "embed" forum posts which are considered "solved". This may include common issues with workaround code or just questions that are answered. By embedding this, we avoid having to curate yet another platform or create some infrastructure around this. But we are definitely open to other ideas as well!

## Goals 

Make the API docs open and accessible for user-contributed content, to cover additional use cases, workarounds, or information that wasn't originally written. This may include:

- Code examples
- Frequently asked questions
- Other questions unrelated to code
- Lower the barrier to join the community (secondary goal)

**Comment by matheusmatos** [comment]
Hi guys, I like that idea.

This can easily be done using [GetStream](https://getstream.io/activity-feeds/) service. It’s just a comments FlatFeed Group Type where each “userID” is a docs page. Each comment is an activity and each activity can have your own reactions: likes, emojis, or more sub comments.

That's the basics. We don't even need server-side code. But it would be nice to have some authentication before commenting. If Expo doesn't have a single sign-on, Firebase is a good catch. We still don't need server-code (and infra), except for a Firebase Cloud Function to return user_token and use in GetStream servers. Depending on adoption, maybe it all would work still inside free quotes of the two services every month.
